src/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+include("list_of_files.cmake")
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS
+    "-g -auto -convert big_endian -assume byterecl -fp-model strict -fpp"
+    )
+  set(fortran_d_flags "-i4 -r8")
+  set(fortran_8_flags "-i8 -r8")
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
+  set(CMAKE_Fortran_FLAGS "-g -fconvert=big-endian -cpp")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-ggdb -Wall")
+  set(fortran_d_flags "-fdefault-real-8")
+  set(fortran_8_flags "-fdefault-integer-8 -fdefault-real-8")
+endif()
+
+set(kinds "4" "8" "d")
+foreach(kind ${kinds})
+  set(lib_name ${PROJECT_NAME}_${kind})
+  set(module_dir "${CMAKE_CURRENT_BINARY_DIR}/include_${kind}")
+
+  # Compiled with preprocessor definition LSIZE=D, not d
+  string(TOUPPER ${kind} kind_definition)
+
+  set(BUILD_FLAGS "${fortran_${kind}_flags}")
+  if(OpenMP_Fortran_FOUND)
+    set(BUILD_FLAGS "${BUILD_FLAGS} ${OpenMP_Fortran_FLAGS}")
+  endif()
```

Comment:
I think we should do this instead.
remove lines 25-27 and replace with:
```
if(OpenMP_Fortran_FOUND)
  target_link_libraries(${lib_name} OpenMP::OpenMP_Fortran)
endif()
```
before the `target_include_directories` line below.